torch/distributed/_checkpoint/metadata.py

```diff
@@ -0,0 +1,47 @@
+from typing import Any, Dict, List, Union, Optional
+import torch
+from dataclasses import dataclass
+import pickle
+from torch.distributed.shard.sharded_tensor import ShardedTensor, ShardedTensorMetadata, ShardMetadata
+
+TENSOR_TYPE = Union[torch.Tensor, ShardedTensor]
+
+@dataclass
+class StorageMetadata:
```

Comment:
All of this will probably change when we actually use flatbuffers for the metadata?